.changelog/5070.cfg.md

```diff
@@ -0,0 +1,105 @@
+Configuration changes
+
+The node now supports a configuration file that holds all the settings
+that were previously available via command-line options.
+The command-line options themselves have been removed in favor of
+the config file.
+
+The config file supports environment variable substitution using
+the `envsubst` package
+([usage information](https://github.com/a8m/envsubst#docs)).
+
+The following options have been removed:
+
+- `consensus.tendermint.mode` in favor of using the global node mode (`mode`).
+
+- `runtime.mode` in favor of using the global node mode (`mode`).
+
+- `worker.client.port` as it is no longer needed.
+
+- `worker.registration.force_register` as it is deprecated.
+
+The following options have been renamed:
+
+- `datadir` to `common.data_dir`.
+
+- `log.file` to `common.log.file`.
+
+- `log.format` to `common.log.format`.
+
+- `log.level` to `common.log.level`.  The log level is now a map of module
+to log level.  Use the `default` module to map the default log level.
+
+- `debug.rlimit` to `common.debug.rlimit`.
+
+- `debug.allow_root` to `common.debug.allow_root`.
+
+- `pprof.bind` to `pprof.bind_address`.
+
+- `consensus.validator` to `consensus_validator`.
+
+- `consensus.tendermint.*` to `tendermint.*`.
```

Comment:
Could we just drop this `tendermint` thing (especially since it will be renamed upstream) from the config and have everything under `consensus`?